x-pack/legacy/plugins/apm/server/lib/settings/agent_configuration/create_agent_config_index.ts

```diff
@@ -7,13 +7,16 @@
 import { InternalCoreSetup } from 'src/core/server';
 import { CallCluster } from '../../../../../../../../src/legacy/core_plugins/elasticsearch';
 import { getApmIndices } from '../apm_indices/get_apm_indices';
+import { getSavedObjectsClient } from '../../helpers/saved_objects_client';
 
 export async function createApmAgentConfigurationIndex(
   core: InternalCoreSetup
 ) {
   try {
     const { server } = core.http;
-    const indices = await getApmIndices(server);
+    const config = server.config();
+    const savedObjectsClient = getSavedObjectsClient(server);
+    const indices = await getApmIndices({ savedObjectsClient, config });
     const index = indices['apm_oss.apmAgentConfigurationIndex'];
     const { callWithInternalUser } = server.plugins.elasticsearch.getCluster(
       'admin'
```

Comment:
@rudolf @ogupte What is the guidelines for using `admin` over `data`? Afaict we only use `getCluster('admin')`

Reply:
This is a legacy API from back when kibana supported tribe nodes. We've removed support for tribe nodes so the distinction should be removed, they're both doing the same thing.

Reply:
https://github.com/elastic/kibana/issues/49870

Reply:
Ahhh.. This is very good to know. I really didn't like that distinction!

Reply:
same